quipucords/api/source/serializer.py

```diff
@@ -335,6 +330,15 @@ def validate_name(name):
     def validate_hosts(hosts):
         """Make sure the hosts list is present."""
         hosts_list = json.loads(hosts)
+        if not isinstance(hosts_list, list):
+            raise ValidationError(_(messages.SOURCE_HOST_MUST_BE_JSON_ARRAY))
+
+        len_list = len(hosts_list)
+        hosts_list = [
+            string for string in hosts_list if isinstance(string, str)]
+        if len_list != len(hosts_list):
+            raise ValidationError(_(messages.SOURCE_HOST_MUST_BE_JSON_ARRAY))
+
```

Comment:
probably should be after the `if not host_list:` check.

Also why do the comprehension with length check vs just iteration through the objects and throwing a validation error if any single item is not a string?
This would save memory and kick out on the first item hit.